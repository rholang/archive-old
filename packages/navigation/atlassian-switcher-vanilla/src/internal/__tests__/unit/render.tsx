import React from 'react';
import { shallow } from 'enzyme';

describe('render', () => {
  const renderMock = jest.fn();
  const unmountComponentAtNodeMock = jest.fn();

  beforeEach(() => {
    renderMock.mockReset();
    unmountComponentAtNodeMock.mockReset();

    jest.resetModules();

    jest.doMock('react-dom', () => ({
      render: renderMock,
      unmountComponentAtNode: unmountComponentAtNodeMock,
    }));
  });

  test('should render the switcher with analytics and intl providers', () => {
    let result;

    renderMock.mockImplementation(content => {
      result = shallow(<div>{content}</div>);
    });

    const { render } = require('../../render');

    const noop = () => {};

    render(
      {
        appearance: 'standalone',
        cloudId: 'some-cloud-id',
        disableCustomLinks: true,
        disableHeadings: true,
        disableRecentContainers: true,
        product: 'opsgenie',
      },
      noop,
      document.createElement('div'),
    );

    expect(result).toMatchSnapshot();
  });

  test('should provide a method to destroy the switcher after rendered', () => {
    const { render } = require('../../render');

    const noop = () => {};
    const div = document.createElement('div');

    const renderedSwitcher = render(
      {
        appearance: 'standalone',
        cloudId: 'some-cloud-id',
        disableCustomLinks: true,
        disableHeadings: true,
        disableRecentContainers: true,
        product: 'opsgenie',
      },
      noop,
      div,
    );

    expect(unmountComponentAtNodeMock).toHaveBeenCalledTimes(0);

    renderedSwitcher.destroy();

    expect(unmountComponentAtNodeMock).toHaveBeenCalledWith(div);
  });
});
