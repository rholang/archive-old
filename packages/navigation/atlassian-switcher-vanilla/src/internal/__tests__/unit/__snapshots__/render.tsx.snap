// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render the switcher with analytics and intl providers 1`] = `
<div>
  <IntlProvider>
    <AnalyticsListener
      channel="*"
      onEvent={[Function]}
    >
      <LoadableComponent
        appearance="standalone"
        cloudId="some-cloud-id"
        disableCustomLinks={true}
        disableHeadings={true}
        disableRecentContainers={true}
        product="opsgenie"
      />
    </AnalyticsListener>
  </IntlProvider>
</div>
`;
